### DISCUSSION TITLE
can we develop robots for delivery food in restaurant indoor ?

### DISCUSSION THREAD

**Discussion by damavand1** [introlab/rtabmap#1126] [General]
hi

Is rtabmap sufficiently capable to perform mapping and localization for a robot inside a building reliably and without issues?

[https://www.youtube.com/watch?v=v-8C-WFOV-4](https://www.youtube.com/watch?v=v-8C-WFOV-4)

Does the robot whose link I sent also use rtabmap or something similar to it?

**Comment by matlabbe** [comment]
With the right sensors, yeah. In this video they are probably using a 2D Lidar-based SLAM/localization approach. Rtabmap can also be configured with a 2D lidar to do the same, though it may appear more complicated to setup rtabmap (though not really) than using a SLAM approach dedicated to 2D lidars. One advantage with rtabmap over pure 2D lidar SLAM is that you can include a camera (that is the part making a little more complex as TF and camera calibration should be right), which will help the robot to globally localize.

Note: When I say right sensors, it means sensors with FOV and Range matching the target environment. I don't think any solution is 100% "reliably and without issues", it is always possible to change the environment to mess-up the localization.
